### DISCUSSION TITLE
Cypress session/origin in headless mode Chromium fails

### DISCUSSION THREAD

**Discussion by rocklee-xyz** [cypress-io/cypress#24632] [Questions and Help] [answered]
I'm facing an issue using `cy.session` and `cy.origin` in Chrome headless mode running in Azure Devops.

Pipeline error occurs in Chrome: 
```
cypress:server:browsers:chrome continueRequest failed url: https://auth.domain.com/oauth2####
```

Login command (OAuth 2 authentication flow with AWS Cognito hosted UI): 

```
Cypress.Commands.add("login", (email, password) => {
  cy.session(
    [email, password],
    () => {
      cy.visit(Cypress.env("baseUrl"));
      cy.origin(
        `${Cypress.env("authUrl")}`,
        { args: [email, password] },
        ([email, password]) => {
          cy.get("input[id='signInFormUsername']").type(email);
          cy.get("input[id='signInFormPassword']").type(password);
          cy.get("input[name='signInSubmitButton']").click();
        }
      );
    },
    {
      validate() {
        cy.visit(Cypress.env("baseUrl"));
      },
      cacheAcrossSpecs: true,
    }
  );
});
```

**To note:**
* Tests are running from a AWS EC2 instance in Azure Devops.
* Tests run in headless mode successfully in Firefox, not Chrome/Edge

Does Chromium require some specific browser preferences?

**Comment by portyu9** [comment, marked as answer]
Hi @rocklee-xyz 

Chromium does not require a special browser preference for the supported `cy.origin()` + `cy.session()` Cognito flow.

This discussion was posted around Cypress 11, while Session/Origin functionality was still evolving. They became generally available in Cypress 12, and current Cypress documentation specifically demonstrates AWS Cognito authentication using `cy.origin()` inside a login flow cached with `cy.session()`.

So I would first reproduce this on a current supported Cypress/browser combination rather than adding Chromium launch flags.

There is also one issue in the example worth fixing:

```js
validate() {
  cy.visit(baseUrl)
}
```

Visiting the page does not actually prove the restored session is authenticated. Validate something that only succeeds when logged in, such as an authenticated API response, cookie/session state, or authenticated UI element.

Also note that on Cypress 16, `Cypress.env()` no longer exists. Current code should use `cy.env()` for sensitive values or `Cypress.expose()` for non-sensitive synchronous configuration.

The 2022 `continueRequest failed` log alone is not enough to prove the original Chromium root cause, so I would not attribute it to browser preferences based on the evidence in the thread.

Please mark this answer as accepted if it helped, thank you.
